### ISSUE TITLE
Coverage reported as 0% when using `cargo test --all`

### ISSUE THREAD

**Issue by euclio** [roblabla/cargo-travis#5]
See report here: https://github.com/Hoverbear/rust-rosetta/pull/563

**Comment by roblabla** [comment]
Looking into it

**Comment by roblabla** [comment]
https://travis-ci.org/Hoverbear/rust-rosetta/builds/205573979#L4329 Cargo coveralls isn't running any tests :|

**Comment by roblabla** [comment]
could you try `cargo coveralls --all` ?

**Comment by euclio** [comment]
Build log with `cargo coveralls --all`: https://travis-ci.org/Hoverbear/rust-rosetta/builds/205580675

**Comment by roblabla** [comment]
So at least this time it ran the builds. The problem is, kcov seems to choke when trying to open some directories. I cloned your repo and managed to reproduce the issue. I'll play around with it and report back when I get it to work :).

**Comment by roblabla** [comment]
So this problem is due to the way workspaces work. The "target" directory isn't next to the cwd() in this case. I'm refactoring the code to use cargo methods to recover the location of the "target" directory. This should be much more robust, and allow it to also support things like `CARGO_TARGET_DIR`

**Comment by roblabla** [comment]
0.0.3 is published on crates.io. If you run the builds again, it should work.

**Comment by euclio** [comment]
Ah, thank you for the quick work! It recorded coverage, but also took the coverage of some library dependencies (...? I see glium, glutin etc. in the covered files). So, it's giving incorrect coverage. Coverage log here: https://coveralls.io/builds/10337936

**Comment by roblabla** [comment]
You're right. It seems that it's more problems with workspaces. I'm investigating.

**Comment by roblabla** [comment]
Fixed with 2581789cc42ee4c893c7694b00978465ff21de34 . 0.0.4 released.

**Comment by roblabla** [comment]
Woops. This wasn't quite fixed, and I forgot to reopen it. 

https://coveralls.io/builds/10347968 It's really weird, coveralls seems eager to include files that are not in the include-path. I need to investigate this more. I'll try to go over it this week-end.
